### PR TITLE
feat(repo): add pipeline type to structs

### PR DESCRIPTION
**Pull request by kneal**
Changes included in PR:

- add new pipeline type field to repo library struct
- add new pipeline type field to repo database struct

Co-authored-by: @JordanSussman

### CODE REVIEW

File: constants/repo.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+// Repo get pipeline types.
+const (
+	// PipelineTypeYAML defines the pipeline type for allowing users
+	// in Vela to control their pipeline being compiled as yaml.
+	PipelineTypeYAML = "yaml"
+
+	// PipelineTypeGo defines the pipeline type for allowing users
+	// in Vela to control their pipeline being compiled as Go templates.
+	PipelineTypeGo = "yaml"
```
Comment (JordanSussman) [inline constants/repo.go:15]: This should be updated to `go` or `golang`.

File: constants/repo.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	// PipelineTypeStarlark defines the pipeline type for allowing users
+	// in Vela to control their pipeline being compiled as Starlark templates.
+	PipelineTypeStarlark = "yaml"
```
Comment (JordanSussman) [inline constants/repo.go:19]: This should be updated to `starlark`.

File: database/repo.go [hunk 3]
```diff
@@ -175,6 +176,11 @@ func (r *Repo) Nullify() *Repo {
 		r.Visibility.Valid = false
 	}
 
+	// // check if the PipelineType field should be false
+	// if len(r.PipelineType.String) == 0 {
+	// 	r.PipelineType.Valid = false
+	// }
```
Comment (JordanSussman) [inline database/repo.go:182]: This should be uncommented.

File: library/repo.go [hunk 4]
```diff
@@ -615,6 +643,7 @@ func (r *Repo) String() string {
   Trusted: %t,
   UserID: %d
   Visibility: %s,
+	PipelineType: %s,
```
Comment (JordanSussman) [inline library/repo.go:646]: Need to fix indentation to align with the other code.

File: library/repo_test.go [hunk 5]
```diff
@@ -295,6 +305,7 @@ func TestLibrary_Repo_String(t *testing.T) {
   Trusted: %t,
   UserID: %d
   Visibility: %s,
+	PipelineType: %s,
```
Comment (JordanSussman) [inline library/repo_test.go:308]: Need to fix indentation to align with the other code.

**Review by JordanSussman — CHANGES_REQUESTED** [review]
initial feedback

**Review by wass3r — APPROVED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by JordanSussman — APPROVED** [review]
LGTM

**Pull request merged**